include/caffe/net.hpp

```diff
@@ -32,11 +32,10 @@ class Net {
   void Init(const NetParameter& param);
 
   /**
-   * @brief Run Forward with the input Blob%s already fed separately.
+   * @brief Run Forward and return the result.
    *
-   * You can get the input blobs using input_blobs().
    */
-  const vector<Blob<Dtype>*>& ForwardPrefilled(Dtype* loss = NULL);
```

Comment:
IMO at least `ForwardPrefilled` should be marked as deprecated and rewritten to just wrap around the new `Forward` -- just removing it breaks Caffe for probably most anyone using it for inference in production.  Not sure if something similar would be reasonable for the other `Forward` methods.